src/frequenz/client/dispatch/test/_service.py

```diff
@@ -178,6 +179,44 @@ def _filter_dispatch(
                 if dispatch.dry_run != _filter.is_dry_run:
                     return False
 
+            if len(_filter.dispatch_ids) > 0 and dispatch.id not in map(
+                DispatchId, _filter.dispatch_ids
+            ):
+                return False
+
+            if not FakeService.matches_query(_filter, dispatch):
+                return False
+
+        return True
+
+    @staticmethod
+    def matches_query(_filter: DispatchFilter, dispatch: Dispatch) -> bool:
```

Comment:
```suggestion
    def matches_query(filter_: DispatchFilter, dispatch: Dispatch) -> bool:
```

Usually a leading `_` is used for *unused* stuff, if you just want to use a keyword or reserved word in any way, a trailing `_` is usually used. If in this case you to shut up pylint, I would add a `pylint: disable=` instead), otherwise the API looks ugly.

Reply:
I probably can make this a private method, too